File: app/src/main/java/es/voghdev/prjdagger2/usecase/GetUsers.java

package es.voghdev.prjdagger2.usecase;

import java.util.List;

import es.voghdev.prjdagger2.global.model.User;

public interface GetUsers {
    List<User> get();

    void getAsync(Listener listener);

    interface Listener {
        void onUsersReceived(List<User> users, boolean isCached);

        void onError(Exception e);

        void onNoInternetAvailable();
    }
}

File: app/src/main/java/es/voghdev/prjdagger2/usecase/GetUsers.kt
package es.voghdev.prjdagger2.usecase

import es.voghdev.prjdagger2.global.model.User

interface GetUsers {
    fun get(): List<User>

    fun getAsync(listener: Listener?)

    interface Listener {
        fun onUsersReceived(users: List<User>, isCached: Boolean)
        fun onError(e: Exception)
        fun onNoInternetAvailable()
    }
}
